File: capability/tools/weather.py
"""Mock weather tool returning canned data."""

from __future__ import annotations

from capability.base import Tool

_MOCK_WEATHER: dict[str, dict] = {
    "tokyo": {"temp": 22, "condition": "Partly cloudy", "humidity": 60},
    "london": {"temp": 14, "condition": "Rainy", "humidity": 80},
    "new york": {"temp": 18, "condition": "Sunny", "humidity": 45},
    "paris": {"temp": 16, "condition": "Overcast", "humidity": 70},
    "sydney": {"temp": 25, "condition": "Clear", "humidity": 55},
}


class WeatherTool(Tool):
    """Return mock weather data for known cities."""

    @property
    def name(self) -> str:
        return "weather"

    @property
    def description(self) -> str:
        return "Get the current weather for a city."

    @property
    def input_schema(self) -> dict:
        return {
            "type": "object",
            "properties": {
                "city": {
                    "type": "string",
                    "description": "City name",
                }
            },
            "required": ["city"],
        }

    async def execute(self, **params) -> str:
        city = params.get("city", "").strip()
        if not city:
            return "Error: city is required"
        data = _MOCK_WEATHER.get(city.lower())
        if data is None:
            return f"No weather data available for '{city}'."
        return (
            f"Weather in {city}: {data['temp']}C, "
            f"{data['condition']}, humidity {data['humidity']}%"
        )

File: src/capability/tools/weather.ts
import { BaseTool } from '../base.js'

/** 天气数据接口 */
interface WeatherData {
  temp: number
  condition: string
  humidity: number
}

/** 模拟天气数据 */
const WEATHER_DATA: Record<string, WeatherData> = {
  beijing: { temp: 22, condition: 'Sunny', humidity: 45 },
  shanghai: { temp: 25, condition: 'Cloudy', humidity: 70 },
  guangzhou: { temp: 30, condition: 'Rainy', humidity: 85 },
  shenzhen: { temp: 29, condition: 'Partly Cloudy', humidity: 78 },
  hangzhou: { temp: 24, condition: 'Overcast', humidity: 65 },
}

/**
 * 天气工具 — 返回指定城市的模拟天气数据
 * 支持: beijing, shanghai, guangzhou, shenzhen, hangzhou
 */
export class WeatherTool extends BaseTool {
  readonly name = 'weather'
  readonly description = 'Get weather information for a city'
  readonly inputSchema = {
    type: 'object',
    properties: {
      city: { type: 'string' },
    },
    required: ['city'],
  }

  async execute(params: Record<string, unknown>): Promise<string> {
    const city = params.city as string
    const data = WEATHER_DATA[city]

    if (!data) {
      return `Weather data not available for ${city}`
    }

    return `${city}: ${data.temp}°C, ${data.condition}, Humidity: ${data.humidity}%`
  }
}
